### PR TITLE
Add LM Eval Testing

### PR DESCRIPTION
**Pull request by dsikka**
# Summary:
- Add LM Eval Teseting to run on a `weekly` cadence
- Model is quantized after which lm_eval runs and the metrics produced are asserted against metrics defined in the config for the particular test case
- Updates the bash script to take in configs folder and test script, allowing it to work for both the vLLM e2e tests and LM Eval tests

# Testing
- Testing has to be updated to use the following commands:
```bash
bash tests/e2e/vLLM/run_tests.sh -c tests/e2e/vLLM/lm_eval_configs -t tests/e2e/vLLM/test_lmeval.py 
bash tests/e2e/vLLM/run_tests.sh -c tests/e2e/vLLM/configs -t tests/e2e/vLLM/test_vllm.py 
```

- LM Eval testing requires `CADENCE=weekly`

### CODE REVIEW

File: setup.py
Comment (robertgshaw2-neuralmagic) [inline setup.py:74]: I would pin a specific version

File: tests/e2e/vLLM/test_lmeval.py
Comment (robertgshaw2-neuralmagic) [inline tests/e2e/vLLM/test_lmeval.py:109]: this will run the model using transformers, not vllm.

File: tests/e2e/vLLM/test_lmeval.py
Comment (robertgshaw2-neuralmagic) [inline tests/e2e/vLLM/test_lmeval.py:115]: with vllm, use `batch_size="auto"`

File: tests/e2e/vLLM/test_lmeval.py
Comment (dsikka) [reply tests/e2e/vLLM/test_lmeval.py:109]: yes. that is the goal. vllm lm_eval is already covered through vllm testing

File: tests/e2e/vLLM/test_lmeval.py
Comment (dsikka) [reply tests/e2e/vLLM/test_lmeval.py:115]: meant for hf

File: tests/e2e/vLLM/run_tests.sh
Comment (dbarbuzzi) [inline tests/e2e/vLLM/run_tests.sh:14]: The `usage` function is not defined anywhere, so this attempt to call it will result in an error that will mask the real error.

Comment (dhuangnm) [reply tests/e2e/vLLM/run_tests.sh:14]: Same as what Domenic mentioned above. I'll update the test workflow to include the lm eval tests once this PR is landed.

File: tests/e2e/vLLM/test_lmeval.py
Comment (robertgshaw2-neuralmagic) [reply tests/e2e/vLLM/test_lmeval.py:109]: my b, I was confused because this is in the `tests/e2e/vLLM` directory

Comment (dsikka) [reply tests/e2e/vLLM/test_lmeval.py:109]: good point

File: tests/e2e/vLLM/run_tests.sh
Comment (dsikka) [reply tests/e2e/vLLM/run_tests.sh:14]: Will update

**Comment by dsikka** [comment]
- To add:
- w4 + act order
- int8 + smoothquant

File: tests/e2e/vLLM/test_lmeval.py
Comment (horheynm) [inline tests/e2e/vLLM/test_lmeval.py:31]: Do you not need to inherit unittest here?

**Comment by horheynm** [comment]
one ocmment otherwise lgtm

File: tests/e2e/vLLM/test_lmeval.py
Comment (dsikka) [reply tests/e2e/vLLM/test_lmeval.py:31]: No, using pyest

Comment (rahul-tuli) [reply tests/e2e/vLLM/test_lmeval.py:31]: We don't need to since pytest can pick up both!

**Review by rahul-tuli — APPROVED** [review]
LGTM pending Domenic's comment about "usage" function in the bash script

File: tests/e2e/vLLM/run_tests.sh
Comment (dhuangnm) [reply tests/e2e/vLLM/run_tests.sh:14]: The code seems not updated yet to include the usage function definition?

**Review by kylesayrs — DISMISSED** [review]

**Pull request merged**
